### PR TITLE
fix(form): fix `useFormItemStatus` get access to window error when sever-side-rendering

### LINKED ISSUE #40976: `useFormItemStatus` 在Next.js服务端渲染时会报错
### Reproduction link

[https://stackblitz.com/edit/nextjs-9tnoqz?file=pages/index.js](https://stackblitz.com/edit/nextjs-9tnoqz?file=pages/index.js)

### Steps to reproduce

在Next.js的需预渲染的组件中中使用 `useFormItemStatu`

### What is expected?

正常使用

### What is actually happening?

Next服务端渲染时提示:Server Error ReferenceError: window is not defined

| Environment | Info |
| --- | --- |
| antd | 5.2.3 |
| React | 18.2.0 |
| System | MacOSX |
| Browser | Chrome 109 |

### PR DESCRIPTION
**Pull request by AndyBoat**
[[English Template / 英文模板](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE.md?plain=1)]

### 🤔 这个变动的性质是？

- [ ] 新特性提交
- [x] 日常 bug 修复
- [ ] 站点、文档改进
- [ ] 演示代码改进
- [ ] 组件样式/交互改进
- [ ] TypeScript 定义更新
- [ ] 包体积优化
- [ ] 性能优化
- [ ] 功能增强
- [ ] 国际化改进
- [ ] 重构
- [ ] 代码风格优化
- [ ] 测试用例
- [ ] 分支合并
- [ ] 工作流程
- [ ] 其他改动（是关于什么的改动？）

### 🔗 相关 Issue
fix #40976

### 💡 需求背景和解决方案

1. 在SSR场景下, useFormItemStatus 对`window`的访问导致SSR渲染失败的问题
2. 参考工程中其他地方,首先判断typeof window是否为`undefined`, 来判断当前是否处于client
3. 只有在客户端, 才会执行warning

### 📝 更新日志

| 语言    | 更新描述 |
| ------- | -------- |
| 🇺🇸 英文fix: fix bug of server error from `useFromItemStatus` in SSR mode         |
| 🇨🇳 中文解决在SSR场景下, useFormItemStatus 对`window`的访问导致SSR渲染失败的问题 |          |

### ☑️ 请求合并前的自查清单

⚠️ 请自检并全部**勾选全部选项**。⚠️

- [x] 文档已补充或无须补充
- [x] 代码演示已提供或无须提供
- [x] TypeScript 定义已补充或无须补充
- [x] Changelog 已提供或无须提供

### CODE REVIEW

**Comment by zombieJ** [comment]
This check seems bug since it does under `Form.Item`. And client check is not best way. 

cc @MadCcc

**Comment by AndyBoat** [comment]
@zombieJ Actually the warning here is not triggered, which is a expected behavior, because the `warning` method will take the first argument (`status !== undefined`) as validator. The reason that casued the error of ssr is the string template, which get access to `window.location`

**Comment by MadCcc** [comment]
> > @zombieJ Actually the warning here is not triggered, which is a expected behavior, because the `warning` method will take the first argument (`status !== undefined`) as validator. The reason that casued the error of ssr is the template literal in the rest arguments of `warning`, trying to read `window.location`.
> 
> Got it. Seems this check using `window` not make sense. It actually just ref to `https://ant.design/components/form-cn#formitemusestatus`. No need check `window` obj. Could you help to replace this with the url directly?

Yeah. Just replace it with online docs link.

**Comment by AndyBoat** [comment]
> 

My pleasue, I will replace it in 24 hours

> > @zombieJ Actually the warning here is not triggered, which is a expected behavior, because the `warning` method will take the first argument (`status !== undefined`) as validator. The reason that casued the error of ssr is the template literal in the rest arguments of `warning`, trying to read `window.location`.
> 
> Got it. Seems this check using `window` not make sense. It actually just ref to `https://ant.design/components/form-cn#formitemusestatus`. No need check `window` obj. Could you help to replace this with the url directly?

My pleasure, I  will make it in 24 hours

**Comment by AndyBoat** [comment]
BTW, wouldn't it be more general to refer to the English doc?
Like what we have done in the `ConfirmDialog` [Line 52](https://github.com/ant-design/ant-design/blob/e56184aaa3c28339ffc2fd4fd7341d17b535cc84/components/modal/ConfirmDialog.tsx#L52)
```js
  warning(
    !(typeof icon === 'string' && icon.length > 2),
    'Modal',
    `\`icon\` is using ReactNode instead of string naming in v4. Please check \`${icon}\` at https://ant.design/components/icon`,
  );
```

**Comment by zombieJ** [comment]
> https://ant.design/components/form-cn#formitemusestatus

You are right. Use en version: https://ant.design/components/form#formitemusestatus

File: components/form/hooks/useFormItemStatus.ts [hunk 1]
```diff
@@ -10,11 +10,14 @@ type UseFormItemStatus = () => {
...
-    status !== undefined,
-    'Form.Item',
-    `Form.Item.useStatus should be used under Form.Item component. For more information: ${window.location.protocol}//${window.location.host}/components/form-cn/#Form.Item.useStatus`,
-  );
+  const isClient = typeof window !== 'undefined'
+  if (isClient) {
+    warning(
+      status !== undefined,
+      'Form.Item',
+      `Form.Item.useStatus should be used under Form.Item component. For more information: ${window.location.protocol}//${window.location.host}/components/form-cn/#Form.Item.useStatus`,
```
Comment (afc163) [inline components/form/hooks/useFormItemStatus.ts]:
```suggestion
      `Form.Item.useStatus should be used under Form.Item component. For more information: https://u.ant.design/form-item-usestatus`,
```

File: components/form/hooks/useFormItemStatus.ts [hunk 2]
```diff
@@ -10,11 +10,14 @@ type UseFormItemStatus = () => {
...
-    status !== undefined,
-    'Form.Item',
-    `Form.Item.useStatus should be used under Form.Item component. For more information: ${window.location.protocol}//${window.location.host}/components/form-cn/#Form.Item.useStatus`,
-  );
+  const isClient = typeof window !== 'undefined'
+  if (isClient) {
+    warning(
+      status !== undefined,
+      'Form.Item',
+      `Form.Item.useStatus should be used under Form.Item component. For more information: https://ant.design/components/form#formitemusestatus,
```
Comment (afc163) [inline components/form/hooks/useFormItemStatus.ts:18]:
```suggestion
      `Form.Item.useStatus should be used under Form.Item component. For more information: https://u.ant.design/form-item-usestatus,
```

File: components/form/hooks/useFormItemStatus.ts [hunk 3]
```diff
@@ -10,11 +10,14 @@ type UseFormItemStatus = () => {
 const useFormItemStatus: UseFormItemStatus = () => {
   const { status } = useContext(FormItemInputContext);
 
-  warning(
-    status !== undefined,
-    'Form.Item',
-    `Form.Item.useStatus should be used under Form.Item component. For more information: ${window.location.protocol}//${window.location.host}/components/form-cn/#Form.Item.useStatus`,
-  );
+  const isClient = typeof window !== 'undefined'
```
Comment (afc163) [inline components/form/hooks/useFormItemStatus.ts:13]: `isClient` is unneccessary now.

**Review by afc163 — APPROVED** [review]

**Pull request merged**